CMakeLists.txt

```diff
@@ -20,57 +20,74 @@ if(STATIC_LIBRARIES)
 endif()
 
 # We use additional modules that cmake needs to know about
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake/Modules/")
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/Modules/" "${PROJECT_SOURCE_DIR}/nuclear/cmake/Modules/")
 
-# Include all of the globally shared link libraries
-include(SharedLibraries)
+# Default to do a debug build
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to 'Debug' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      Debug
+      CACHE STRING "Choose the type of build." FORCE
+  )
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
 
-# Include specific compiler options
-include(Compiler)
+# Default not to run the clang-tidy checks
+option(ENABLE_CLANG_TIDY "Enable building with clang-tidy checks." OFF)
+if(ENABLE_CLANG_TIDY)
+  set(CMAKE_CXX_CLANG_TIDY clang-tidy)
+  set(CMAKE_C_CLANG_TIDY clang-tidy)
+endif(ENABLE_CLANG_TIDY)
 
-# Set flag so NUClear uses a custom clock implementation
-option(USE_CUSTOM_CLOCK "Replace NUClear's clock with a custom implementation." ON)
-if(USE_CUSTOM_CLOCK)
-  add_definitions(-DNUCLEAR_CUSTOM_CLOCK)
-  add_subdirectory(shared/clock)
-endif(USE_CUSTOM_CLOCK)
+# Enable address sanitizer
+option(USE_ASAN "Enable address sanitization" OFF)
+if(USE_ASAN)
+  add_compile_options(-fsanitize=address -fno-omit-frame-pointer -U_FORTIFY_SOURCE -fno-common)
+  add_link_options(-fsanitize=address)
+  link_libraries(asan)
+endif(USE_ASAN)
 
-# Hide some cmake variables that will never be used
-mark_as_advanced(
-  CMAKE_INSTALL_PREFIX
-  CURSES_CURSES_H_PATH
-  CURSES_FORM_LIBRARY
-  CURSES_HAVE_CURSES_H
-  OMPLAPP_LIBRARY
-  OMPL_CONFIG
-  OMPL_DIR
-  OMPL_INCLUDE_DIRS
-  OMPL_LIBRARIES
-  OMPL_LIBRARY
+# Enable testing if we are building tests
+if(BUILD_TESTS)
+  enable_testing()
+  add_subdirectory(shared/tests)
+endif(BUILD_TESTS)
```

Comment:
Move this to the end of the file, this should fix the `${NUCLEAR_SHARED_DIR}` problem we are seeing in `shared/tests`

Reply:
Yep that worked. :grin: